### DISCUSSION TITLE
Pyright should narrow `except*` to ExceptionGroup with non-base exceptions

### DISCUSSION THREAD

**Discussion by jakkdl** [microsoft/pyright#9455] [General]
It was changed from `ExceptionGroup` to `BaseExceptionGroup` in https://github.com/microsoft/pylance-release/issues/4183
Related typeshed issue https://github.com/python/typeshed/issues/12972

I'm not 100% if the typeshed issue will resolve `except*` handling in pyright, in case it would then we can directly resolve/close this. But in case it's hard-coded to always be `BaseExceptionGroup` and won't be changed by typeshed messing with `__new__` then you might want to fix it right away.

repro:
```python
from typing import reveal_type
try:
    raise ExceptionGroup("", [ValueError()])
except* ValueError as e:
    reveal_type(e)
```

```sh
$ python repro.py
Runtime type is 'ExceptionGroup'
$ pyright repro.py
  repro.py:5:17 - information: Type of "e" is "BaseExceptionGroup[ValueError]"
0 errors, 0 warnings, 1 information 
```

**Comment by erictraut** [comment]
I'm not sure what you mean. Could you provide a minimal code sample that demonstrates the behavior that you're seeing and your expected behavior?

**Reply by jakkdl** [reply]
ah sorry, added to OP

**Reply by erictraut** [reply]
I've created an enhancement request to track this: https://github.com/microsoft/pyright/issues/9466.
